fog/view/server/src/block_tracker.rs

```diff
@@ -72,14 +71,33 @@ impl BlockTracker {
 
     /// Given a list of ingestable ranges, missing blocks and current state, calculate the highest
     /// processed block count number. The highest processed block count number is the block count
-    /// for which we know we have loaded all available data.
+    /// for which we know we have loaded all required data, so the users can potentially compute
+    /// their balance up to this block without missing any transactions.
+    ///
+    /// Arguments:
+    /// * highest_known_block_index:
+    ///   The highest block index known to have appeared in the blockchain.
+    ///   If there are no ingress keys at all right now, then this is also the highest
+    ///   fully-processed block. It is assumed in the algorithm that any new keys will
+    ///   appear with start block after this number.
+    /// * ingress_keys:
+    ///   IngressPublicKeyRecord's that exist in the database right now.
+    ///   This indicates their start block, their last-scanned block, their expiry block,
+    ///   and whether they are retired.
+    /// * missing_block_ranges:
+    ///   Any manually entered missing block ranges.
+    ///
```

Comment:
I need to add a code comment explaining the return value:

The idea is, we return what the highest fully processed block count is, and, if an Ingress key is what is blocking further progress, a copy of that record. So that if we are blocked on the same thing for a long time we can log